roles/container-engine/containerd/defaults/main.yml

```
@@ -109,3 +109,11 @@ containerd_supported_distributions:
 
 # Enable container device interface
 enable_cdi: false
+
+# For containerd tracing configuration please check out the official documentation:
+# https://github.com/containerd/containerd/blob/main/docs/tracing.md
+containerd_tracing_enabled: false
+containerd_tracing_endpoint: "0.0.0.0:4317"
```

The OpenTelemetry protocol port should be `4318` according to https://github.com/containerd/containerd/blob/main/docs/tracing.md
```suggestion
containerd_tracing_endpoint: "0.0.0.0:4318"
```

---

Hi @mzaian! I think it has been set to 4318 because it uses http/protobuf as default; it also supports grpc (and usually opentelemetry collector uses 4317 port [doc](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/#otel_exporter_otlp_endpoint)). I configured it to grpc protocol as default just to make it the same as with other components like etcd-apiserver-kubelet ones.

---

Hi @ugur99 , 
Thanks for the explanation. Totally agree with your approach.